### PR TITLE
Students application cont

### PR DESCRIPTION
**Pull request by majakudlicka**
As per commit messages

### CODE REVIEW

File: react-ui/src/actions/job_apply.js [hunk 1]
```diff
@@ -7,7 +7,9 @@ export const submitJobApplication = job_id => dispatch => {
     status: 'pending'
   });
   axios
-    .get('/api/apply')
+    .get('/api/apply', {
```
Comment (des-des) [inline react-ui/src/actions/job_apply.js]: I dont think this should be a get request, get requests should not change anything on the server

File: server/database/db_queries.js [hunk 2]
```diff
@@ -119,6 +119,20 @@ data.postResidents = (resident, callback) => {
     );
 };
 
+data.sendApplication = (job_id, callback) => {
```
Comment (des-des) [inline server/database/db_queries.js]: `submitApplication` better name?

File: server/index.js [hunk 3]
```diff
@@ -109,6 +109,7 @@ server.register(plugins, err => {
         }
     });
 
+    //unhardcode the job id
     server.route({
         method: 'GET',
```
Comment (des-des) [inline server/index.js]: This should not be a get request,

Make it a PUT/POST request with the jobId in the payload

File: server/index.js [hunk 4]
```diff
@@ -118,13 +119,20 @@ server.register(plugins, err => {
...
+                    );
                 } else {
-                    reply({
-                        message: 'Email sent!'
+                    sendEmail(from, to, subject, text, (err, res) => {
+                        if (err) {
+                            reply('Failed to send email').code(500);
+                        } else {
+                            reply({
+                                message: 'Email sent!'
```
Comment (des-des) [inline server/index.js]: add `name` as field to the payload here.

File: server/index.js [hunk 5]
```diff
@@ -118,13 +119,20 @@ server.register(plugins, err => {
             var subject = 'New job application';
             var text =
                 'Someone has applied for the job you posted. Go to your profile to find out more.';
-
-            sendEmail(from, to, subject, text, (err, res) => {
+            data.sendApplication(request.url.query.job_id, (err, res) => {
                 if (err) {
-                    reply('Failed to send email').code(500);
+                    reply('Failed to connect load data from the database').code(
```
Comment (des-des) [inline server/index.js]: see the way @antoniotrkdz is handling errors. You should be using Boom, this is inconsistent .

File: server/index.js (hunk 4 above)
Comment (des-des) [inline server/index.js]: Not having this is inconsistent with the api

File: react-ui/src/actions/job_apply.js (hunk 1 above)
Comment (majakudlicka) [inline react-ui/src/actions/job_apply.js]: ok

File: server/index.js (hunk 3 above)
Comment (majakudlicka) [inline server/index.js]: Yes

File: server/index.js (hunk 5 above)
Comment (majakudlicka) [inline server/index.js]: Is it ok if I make it on a separate branch? Cleaning up our error handling in server is something I intended to do for a while

Comment (majakudlicka) [inline server/index.js]: OK

File: server/index.js [hunk 6]
```diff
@@ -55,6 +55,7 @@ server.register(plugins, err => {
         method: 'GET',
         path: '/api/jobs',
         handler: (request, reply) => {
+          console.log('inside server')
```
Comment (des-des) [inline server/index.js]: Remove this

**Comment by des-des** [comment]
image

latest change (meant to be removing a single console.log) has ~600 lines of changes

**Comment by majakudlicka** [comment]
OK, apart from one merge conflict, looks like the differences were caused by different trailing comma settings between me and A. I have aligned my comma settings to match his (but that means some of my old files may now have extra commas in them). Open for alternative suggestions.

File: server/index.js [hunk 7]
```diff
@@ -64,16 +65,17 @@ server.register(plugins, err => {
                     message: 'Welcome to BEEVR!',
                     jobsList: res.map(element => {
                         return {
-                            jobId: element.job_id,
-                            startDate: element.start_date,
-                            startTime: element.start_time,
-                            endDate: element.end_date,
-                            endTime: element.end_time,
+                            job_id: element.job_id,
```
Comment (des-des) [inline server/index.js]: Okay cool, even though your db is snake_case, your json api should speak camelCase

**Comment by des-des** [comment]
cool. If you realise your and @antoniotrkdz linters are out of sync you should
  1. Resolve the problem
  2. Make a pull request that makes your code base conform to the linting.

This pull should contain not functional changes

**Comment by majakudlicka** [comment]
OK done. Yes, I'd attempt to address the eslint issue in a separate PR

**Pull request merged**